File: filetemplete/Custom/PowerTableViewCell.xctemplate/UITableViewCellXIBObjective-C/___FILEBASENAME___.m
//___FILEHEADER___

#import "___FILEBASENAME___.h"

/* System */

/* ViewController */

/* View */

/* Model */

/* Util */

/* NetWork InterFace */

/* Vender */

@interface ___FILEBASENAMEASIDENTIFIER___ ()

@end
@implementation ___FILEBASENAMEASIDENTIFIER___

#pragma mark - life cycle
- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier
{
    if (self = [super initWithStyle:style reuseIdentifier:reuseIdentifier]) {
        [self commonInit];
    }
    return self;
}

- (instancetype)initWithCoder:(NSCoder *)aDecoder
{
    if (self = [super initWithCoder:aDecoder]) {
        [self commonInit];
    }
    return self;
}

#if DEBUG
- (void)dealloc
{
    NSLog(@"[%@ dealloc]", NSStringFromClass(self.class));
}
#endif

#pragma mark - initial Methods
- (void)commonInit
{
    /*custom view u want draw in here*/
    self.contentView.backgroundColor = [UIColor whiteColor];
}


#pragma mark - layout
- (void)updateConstraints
{
    /*更新约束，自定义view应该重写此方法在其中建立constraints.*/
    
    [super updateConstraints];//this line must be the last line of this method.^_^
}

- (void)layoutSubviews
{
    [super layoutSubviews];
    /*如果你需要更精确控制子view，而不是使用限制或autoresizing行为，就需要实现该方法。*/
    
}

#pragma mark - private method

#pragma mark - public method

#pragma mark - getters and setters
@end

File: filetemplete/Custom/PowerTableViewCell.xctemplate/UITableViewCellXIBSwift/___FILEBASENAME___.swift
//___FILEHEADER___

/* System */
import UIKit

/* ViewController */

/* View */

/* Model */

/* Util */

/* NetWork InterFace */

/* Vender */

class ___FILEBASENAMEASIDENTIFIER___: ___VARIABLE_cocoaTouchSubclass___ {

    //MARK: - public property
    
    //MARK: - readonly property
    
    //MARK: - getters and setters
    
    
    //MARK: - life cycle
    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)

        self.commonInit()
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        
        self.commonInit()
    }
    
    //MARK: - layout
    override func updateConstraints() {
        
        /*更新约束，自定义view应该重写此方法在其中建立constraints.*/
        
        super.updateConstraints() //this line must be the last line of this method.^_^
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        
        /*如果你需要更精确控制子view，而不是使用限制或autoresizing行为，就需要实现该方法。*/
    }

    #if DEBUG
    deinit {
        print("[\(type(of: self)) deinit]")
    }
    #endif
}


//MARK: - initial Methods
fileprivate extension ___FILEBASENAMEASIDENTIFIER___ {
    func commonInit() {
        self.contentView.backgroundColor = UIColor.white;
    }
}

//MARK: - private method
fileprivate extension ___FILEBASENAMEASIDENTIFIER___ {
    
}

//MARK: - public method
extension ___FILEBASENAMEASIDENTIFIER___ {
    
}
